server.js

```diff
@@ -1,30 +1,124 @@
 import express from 'express'
-import bodyParser from 'body-parser'
 import cors from 'cors'
+import dotenv from 'dotenv'
 import mongoose from 'mongoose'
+import listEndpoints from 'express-list-endpoints'
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
+dotenv.config()
+
+// const mongoUrl = process.env.MONGO_URL || "mongodb://localhost:27017/happyThoughts"
+const mongoUrl = process.env.MONGO_URL || `mongodb+srv://${process.env.USER_ID}:${process.env.API_KEY}@cluster0.bvwog.mongodb.net/happyThoughts?retryWrites=true&w=majority`
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useCreateIndex: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
+const port = process.env.PORT || 8081
 const app = express()
 
+const thoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    required: [true, 'Write something, not less than 5 characters'],
+    unique: true,
+    trim: true,
+    minlength: 5, 
+    maxlength: 140,
+    // enum: ['']
+    //match: regex
+  },
+  hearts: {
+    type: Number,
+    default: 0
+  }, 
+  createdAt: {
+    type: Date,  
+    default: Date.now
+  }
+})
+
+//Model for mongo database 
+const Thought = mongoose.model('Thought', thoughtSchema)
+
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
-// Start defining your routes here
+//GET route
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+app.get('/thoughts', async (req, res) => {
+  // const { page, per_page } = req.query
+ 
+  const thoughts = await Thought.find().sort({createdAt: 'desc'}).limit(20).exec()
+  res.json(thoughts)
+})
+
+// POST route
+app.post('/thoughts', async (req, res) => {
+  try {
+    const newThought = await new Thought ({
+      message: req.body.message}).save() 
```

Comment:
If you would like to make the code even shorter here, you could go with just await new Thought(req.body).save() here since it's only the message being included in the req body at the moment.